### PR TITLE
`pack builder create` should support extensions

### PR DESCRIPTION
**Pull request by natalieparellano**
## Summary
If a builder.toml contains `[extensions]` and `[order-extensions]`:
* See https://github.com/buildpacks/spec/pull/316 for detailed requirements

## Documentation

- Should this change be documented?
    - [x] Yes, see https://github.com/buildpacks/docs/issues/501
    - [ ] No

## Related

Resolves https://github.com/buildpacks/pack/issues/1468

### CODE REVIEW

File: pkg/buildpack/buildpack.go
Comment (jromero) [inline pkg/buildpack/buildpack.go:37]: Are we able to remove the `Module` prefix to the getter methods? They fill a tad awkward.

**Review by jromero — COMMENTED** [review]
Initial feedback:

I like the term module overall for the generic shared concept. There are still a few references where “buildpacks” are used when talking about extensions though. I think they need to be fully resolved.

ie.
- BuildpackLayers
- `/pkg/buildpack` package

File: pkg/buildpack/buildpack.go
Comment (natalieparellano) [reply pkg/buildpack/buildpack.go]: Updated in bd37213a

**Comment by natalieparellano** [comment]
Still to be done:
* fix use of deprecated strings.Title
* test filling in extension version from the provided order for extensions
* Ensure `optional = true` isn’t redundantly printed in the order for extensions

**Comment by natalieparellano** [comment]
I'm going to mark this as ready for review as I am pretty confident that once the WCOW worker comes back online, Codecov will be happy.

I'm really sorry about all the changed files and (due to package rename) undecipherable diffs. I'm happy to pair on the code review if that would be helpful.

**Comment by jromero** [comment]
@natalieparellano I've updated the description to define blockers. We'll want the specs released before we can merge this PR. Let me know if that's a concern.

**Comment by jromero** [comment]
As discussed in WG today, we can work around needing to wait for the spec releases if we were to:

1. gate this functionality behind "experimental"
2. remove the statement that platform API 0.10 is supported

**Review by jromero — DISMISSED** [review]
@natalieparellano correct me if I'm wrong, but we're still waiting on this to be updated based on the previous comment.

**Comment by natalieparellano** [comment]
@jromero you are correct - apologies for letting this one go dormant. I'll circle back to it soon...

**Comment by natalieparellano** [comment]
@jromero I made the changes requested. Assuming this passes CI could you PTAL? 🙏🏼

**Comment by natalieparellano** [comment]
It seems that something about the changes make layer diff IDs non-deterministic on Windows. At least, asserting on the contents of `io.buildpacks.extension.layers` is failing, and [this action run](https://github.com/buildpacks/pack/actions/runs/3085654321) shows that layer diff IDs can be [predicted](https://github.com/buildpacks/pack/pull/1516/files) with the code on `main`. I'm a bit surprised that no other tests are failing... will need to dig into it more.

**Comment by natalieparellano** [comment]
Also need to fix compat suite

File: .github/workflows/build.yml
Comment (natalieparellano) [inline .github/workflows/build.yml:107]: Fixes unrelated issue; the existing Powershell was too brittle.

File: pkg/buildpack/buildpack.go
Comment (jkutner) [inline pkg/buildpack/buildpack.go:144]: Should we make `"extension"` a const somewhere? I see it used a few times

**Review by jkutner — APPROVED** [review]

File: pkg/buildpack/buildpack.go
Comment (natalieparellano) [reply pkg/buildpack/buildpack.go]: Updated in [a257564](https://github.com/buildpacks/pack/pull/1478/commits/a257564cf8b45a62ca93ce90e011025d9c4c05c8)

**Review by jjbustamante — APPROVED** [review]
It looks good to me

**Pull request merged**
